src/etc/inc/system.inc

```diff
@@ -616,25 +616,26 @@ function system_dhcpleases_configure() {
 			$unbound_conf = "";
 		}
 
-		if (isvalidpid($pidfile)) {
-			/* Make sure dhcpleases is using correct unbound or dnsmasq */
-			$_gb = exec("/bin/pgrep -F {$pidfile} -f {$dns_pid}", $output, $retval);
-			if (intval($retval) == 0) {
-				sigkillbypid($pidfile, "HUP");
-				return;
-			} else {
-				sigkillbypid($pidfile, "TERM");
-			}
-		}
+		/* best to never send a HUP to dhcpleases, it is bugged */
+		// if (isvalidpid($pidfile)) {
+		// /* Make sure dhcpleases is using correct unbound or dnsmasq */
+		//	$_gb = exec("/bin/pgrep -F {$pidfile} -f {$dns_pid}", $output, $retval);
+		//	if (intval($retval) == 0) {
+		//		sigkillbypid($pidfile, "HUP");
+		//		return;
+		//	} else {
+		//		sigkillbypid($pidfile, "TERM");
+		//	}
+		// }
 
 		/* To ensure we do not start multiple instances of dhcpleases, perform some clean-up first. */
-		if (is_process_running("dhcpleases")) {
-			sigkillbyname('dhcpleases', "TERM");
-		}
+		mwexec("/bin/pkill -x dhcpleases", true, true, false);
```

Comment:
There isn't much to be gained by going this route. sigkillbyname() does its own check if a process is running first but it's a fairly fast pgrep followed by `killall`, which is roughly the same as `pkill -x`.

The only change needed is to remove the router is_process_running() since sigkillbyname() already does that. And though sigkillbyname does use mwexec itself, we prefer to use functions for those where possible so that if we find a better way to implement that action, we don't need to hunt down all of the shell execs.